packages/math/src/Matrix.ts

```diff
@@ -15,6 +16,15 @@ import { PI_2 } from './const';
  */
 export class Matrix
 {
+    public a: number;
+    public b: number;
+    public c: number;
+    public d: number;
+    public tx: number;
+    public ty: number;
+
+    public array: Float32Array|null;
+
```

Comment:
I think we need to make a ruling on how we plan to make these classes. There are a few different things we need to make a decision about and enforce for consistency across the codebase:

1. Do you assign a value to the member in the member declaration, rather than in the ctor?
2. Do you put the docs on the member declaration and not in the ctor?
3. Do you use the constructor param shorthand for members that are the same type/name as a param?
4. Do you always specify function return values, even when they are obvious or void?

My recommendations:

1. Always prefer assignment in the member declaration rather than the ctor, unless the initial value is based on a ctor param.
2. Put the docs on the member declaration, not in the ctor.
3. Do not use constructor param shorthand, it can make documentation blocks a bit messy.
4. Always specify return values, it can ensure you don't accidentally change the implicit return type.

Another note: There is no need to put `@member` or `@default` or `@type` in the docs, they are inferred from the Typescript code itself, and keeping the docs in sync with the code only becomes more difficult when you have to declare these values in two spots.

Here is how this would look with my recommendations:

```ts
class Matrix
{
    public a: number;
    public b: number;
    public c: number;
    public d: number;
    public tx: number;
    public ty: number;

    /**
     * Member Description
     * Assigned to null because the initial value is not based on a param.
     */
    public array: Float32Array | null = null;

    constructor(a = 1, b = 0, c = 0, d = 1, tx = 0, ty = 0)
    {
        this.a = a;
        this.b = b;
        this.c = c;
        this.d = d;
        this.tx = tx;
        this.ty = ty;
    }
}
```

Reply:
So with jsdoc if the documentation isnt inside the constructor and above in the member declaration then it gets stripped out. 

so if we change this:
```js
public a: number
constructor(a = 1)
{
    /**
    * @member {number}
    * @default 1
    */
    this.a = a;
}
```
to this:
```js
/**
* @member {number}
* @default 1
*/
public a: number

constructor(a = 1)
{
    this.a = a;
}
```
this docs are removed.

Unless anyone knows how to stop this from happening (@bigtimebuddy?)  i dont think we move the docs.

Reply:
That's really unfortunate because I'd expect that most members aren't even listen in the ctor, and only in the declaration area. I wonder if instead of using jsdoc on the typescript, we should use typedoc on the javascript.

Reply:
`array` was never documented. However if i put this:

```js
    /**
     * @member {PIXI.Matrix}
     */
    public array: Float32Array|null = null;
```
since its not being used in the constructor then the documentation is kept

Reply:
Ugh! That’s frustrating but I should’ve expected because I know TS removes comments before type declarations even if removeComments is false. Hmm... need to think about this. Maybe we make an initial minimal pass at all the packages to TS and do cleanup after the logic is over?